app/models/nomis/sentence_detail.rb

```diff
@@ -27,6 +29,14 @@ def conditional_release_date
       @conditional_release_override_date.presence || @conditional_release_date
     end
 
+    def recall_release_date
+      if @actual_parole_date.presence && @actual_parole_date < @post_recall_release_override_date
+        @actual_parole_date
+      else
+        @post_recall_release_override_date.presence || @post_recall_release_date
+      end
+    end
+
```

Comment:
I found this conditional a little hard to read. I played with it a little locally to try and understand why.

- I think checking if the actual parole date's presence works better as a guard clause
- That shape forces you to repeat the `else` branch comparison, which made me want in a separate branch

I ended up here, if any of it's useful:
```ruby
def recall_release_date
  return post_recall_release_date if @actual_parole_date.blank?

  if @post_recall_release_override_date > @actual_parole_date
    @actual_parole_date
  else
    post_recall_release_date
  end
end

# is there a better name for this method?
def post_recall_release_date
  @post_recall_release_override_date.presence || @post_recall_release_date
end

```

Reply:
Yes this is so much clearer.